util/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/AttributesUtil.java

```diff
@@ -0,0 +1,219 @@
+package com.google.cloud.healthcare.imaging.dicomadapter;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.dcm4che3.data.Attributes;
+import org.dcm4che3.data.Sequence;
+import org.dcm4che3.data.Tag;
+import org.dcm4che3.data.VR;
+import org.dcm4che3.net.Status;
+import org.dcm4che3.net.service.DicomServiceException;
+import org.dcm4che3.util.TagUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class AttributesUtil {
+
+  public static String getTagValue(JSONObject json, String tag) throws JSONException {
+    JSONObject jsonTag = json.getJSONObject(tag);
+    JSONArray valueArray = jsonTag.getJSONArray("Value");
+    if (valueArray.length() != 1) {
+      throw new JSONException(
+          "Expected single value for tag: "
+              + tag
+              + " in JSON:\n"
+              + valueArray.toString());
+    }
+    return valueArray.getString(0);
+  }
+
+  public static String getTagValueOrNull(JSONObject json, String tag) {
+    if (!json.has(tag)) {
+      return null;
+    }
+    JSONObject jsonTag = json.getJSONObject(tag);
+    if (!jsonTag.has("Value")) {
+      return null;
+    }
+    JSONArray valueArray = jsonTag.getJSONArray("Value");
+    if (valueArray.length() != 1) {
+      return null;
+    }
+    return valueArray.getString(0);
+  }
+
+  /**
+   * Returns array of QIDO-RS paths (more than 1 if multiple ModalitiesInStudy are present)
+   * @param attrs dcm4che Attributes to convert
+   * @param includeFields additonal includeFields for QIDO-RS
+   */
+  public static String[] attributesToQidoPathArray(Attributes attrs, String... includeFields)
+      throws IOException {
+    if (!attrs.containsValue(Tag.ModalitiesInStudy)
+        || attrs.getStrings(Tag.ModalitiesInStudy).length <= 1) {
+      return new String[]{attributesToQidoPath(attrs, includeFields)};
+    }
+
+    List<String> result = new ArrayList<>();
+    for (String modality : attrs.getStrings(Tag.ModalitiesInStudy)) {
```

Comment:
Instead of doing this, can't you add them as comma separated instead? http://dicom.nema.org/medical/dicom/current/output/html/part18.html#sect_10.6.1.2.1
See the last example here. That way you only need to ever make one qido request. Please test it against the CHC API to verify it works.

Reply:
Comma-separated notation within single parameter doesn't seem to be recognized by healthcare api at all (always empty response), as far as I remember dcm4che took first modality instead.

Repeating same parameter is at least recognized syntax, but returns the following: 
"error": {
            "code": 400,
            "message": "generic::invalid_argument: list of UIDs matching unsupported, tag: ModalitiesInStudy",
            "status": "INVALID_ARGUMENT"
        }

And I get same error for StudyInstanceUID despite the fact that it is used in standard example.

I also forwarded you previous e-mail chain I had on this matter with Dusan.

Reply:
Fine to leave this as-is.